src/main/scala/io/snyk/plugin/ui/settings/SettingsDialog.scala

```diff
@@ -0,0 +1,198 @@
+package io.snyk.plugin.ui.settings
+
+import java.net.URL
+
+import com.intellij.openapi.ui.{ComponentValidator, ValidationInfo}
+import javax.swing.{JCheckBox, JComponent, JLabel, JPanel, JTextField}
+import java.awt.{Dimension, Insets}
+
+import com.intellij.openapi.project.Project
+import com.intellij.ui.DocumentAdapter
+import com.intellij.uiDesigner.core.{GridConstraints => UIGridConstraints, GridLayoutManager => UIGridLayoutManager}
+import com.intellij.uiDesigner.core.Spacer
+import javax.swing.event.DocumentEvent
+
+class SettingsDialog(project: Project) {
+
+  private val persistentState: SnykPersistentStateComponent = SnykPersistentStateComponent.getInstance(project)
+
+  private val customEndpointTextField: JTextField = new JTextField()
+  private val organizationTextField: JTextField = new JTextField()
+  private val ignoreUnknownCACheckBox: JCheckBox = new JCheckBox()
+
+  private val rootPanel: JPanel = new JPanel()
+
+  setupUI()
+  setupValidation()
+
+  if (persistentState != null) {
+    reset()
+  }
+
+  def getRootPanel: JComponent = rootPanel
+
+  def apply(): Unit = {
+    val customEndpoint = customEndpointTextField.getText
+
+    if (!isUrlValid(customEndpoint)) {
+      return
+    }
+
+    persistentState.customEndpointUrl = customEndpoint
+    persistentState.organization = organizationTextField.getText
+    persistentState.ignoreUnknownCA = ignoreUnknownCACheckBox.isSelected
+  }
+
+  def reset(): Unit = {
+    customEndpointTextField.setText(persistentState.getCustomEndpointUrl())
+    organizationTextField.setText(persistentState.getOrganization())
+    ignoreUnknownCACheckBox.setSelected(persistentState.isIgnoreUnknownCA())
+  }
+
+  def isModified: Boolean =
+     isCustomEndpointModified || isOrganizationModified || isIgnoreUnknownCAModified
+
+  private def isCustomEndpointModified =
+    customEndpointTextField.getText != persistentState.getCustomEndpointUrl()
+
+  private def isOrganizationModified =
+    organizationTextField.getText() != persistentState.getOrganization()
+
+  private def isIgnoreUnknownCAModified =
+    ignoreUnknownCACheckBox.isSelected != persistentState.isIgnoreUnknownCA()
+
+  private def setupUI(): Unit = {
+    val gridLayoutManager = new UIGridLayoutManager(5, 2, new Insets(0, 0, 0, 0), -1, -1)
+
+    rootPanel.setLayout(gridLayoutManager)
+
+    val customEndpointLabel = new JLabel
+    customEndpointLabel.setText("Custom Endpoint:")
```

Comment:
Should it be "Custom endpoint:" without capitalization the second word?
I checked UI guidelines from JetBrains about capitalization https://jetbrains.github.io/ui/text/capitalization/#sentence and I think this is our case.

Reply:
Good point @pavel-snyk 👍